src/core/qgsvectorlayerlabelprovider.cpp

```diff
@@ -95,6 +95,12 @@ QgsVectorLayerLabelProvider::QgsVectorLayerLabelProvider( const QgsPalLayerSetti
 
 void QgsVectorLayerLabelProvider::init()
 {
+  if ( mLayerGeometryType == QgsWkbTypes::GeometryType::PolygonGeometry &&
```

Comment:
I think we should leave off that check, and instead split the polygon perimeter option from reusing "Line" placement to having it's own "PerimeterStraight" placement enum value. This would help with other issues - because pal has no knowledge of the original geometry for perimeter labels but it would be useful to differentiate between perimeter based straight labels and standard straight line labels...